app/scripts/modules/core/src/serverGroup/ServerGroup.tsx

```diff
@@ -0,0 +1,272 @@
+      const fromFullUrl = jenkinsConfig.fullUrl;
+      const fromBuildInfo = serverGroup.buildInfo.buildInfoUrl;
+
+      jenkins = {
+        number: jenkinsConfig.number,
+        href: fromHost || fromFullUrl || fromBuildInfo,
+      };
+    } else if (has(serverGroup, 'buildInfo.images')) {
+      images = serverGroup.buildInfo.images.join(', ');
+    }
+
+    return {
+      serverGroup,
+      serverGroupSequence,
+      jenkins,
+      hasBuildInfo,
+      instances,
+      images,
+      filter,
+      showAllInstances,
+      listInstances,
+      multiselect,
+      isSelected,
+      isMultiSelected,
+    };
+  }
+
+  private isSelected(serverGroup: IServerGroup) {
+    const params = {
+      region: serverGroup.region,
+      accountId: serverGroup.account,
+      serverGroup: serverGroup.name,
+      provider: serverGroup.type,
+    };
+
+    return ReactInjector.$state.includes('**.serverGroup', params);
+  }
+
+  private isMultiSelected(multiselect: boolean, serverGroup: IServerGroup) {
+    return multiselect && ReactInjector.MultiselectModel.serverGroupIsSelected(serverGroup);
+  }
+
+  private onServerGroupsChanged() {
+    let isMultiSelected = this.isMultiSelected(this.state.multiselect, this.state.serverGroup);
+    this.setState({ isMultiSelected });
+    // Enables the (angular) details pane to detect the changes
+    ReactInjector.$rootScope.$applyAsync(() => false);
+  }
+
+  private onStateChanged() {
+    this.setState({ isSelected: this.isSelected(this.state.serverGroup) });
+  }
+
+  public componentDidMount(): void {
+    const { serverGroupsStream, instancesStream } = ReactInjector.MultiselectModel;
+
+    this.serverGroupsSubscription = serverGroupsStream.merge(instancesStream).subscribe(this.onServerGroupsChanged);
+    this.stateChangeSubscription = ReactInjector.$uiRouter.globals.success$.subscribe(this.onStateChanged);
+    this.onStateChanged();
+  }
+
+  public componentWillUnmount() {
+    this.stateChangeSubscription.unsubscribe();
+    this.serverGroupsSubscription.unsubscribe();
+  }
+
+  public componentWillReceiveProps(nextProps: IServerGroupProps) {
+    this.setState(this.getState(nextProps));
+  }
+
+  public headerIsSticky(): boolean {
+    const { showAllInstances, listInstances, instances } = this.state;
+    if (!showAllInstances) {
+      return false;
+    }
+
+    return listInstances ? instances.length > 1 : instances.length > 20;
+  }
+
+  public loadDetails(event: React.MouseEvent<any>): void {
+    event.persist();
+
+    setTimeout(() => {
+      if (event.isDefaultPrevented() || event.nativeEvent.defaultPrevented) {
+        return;
+      }
+      ReactInjector.MultiselectModel.toggleServerGroup(this.props.serverGroup);
+      event.preventDefault();
+    })
+  };
+
+  private handleServerGroupClicked(event: React.MouseEvent<any>) {
+    this.fireGaEvent();
```

Comment:
Seems odd to have this as a separate generically named function if it's only called here and it's a specific GA event. Either just get rid of the function or rename it.